### PR TITLE
[Add & Modify Units] Udongein, SpellCaster: support various utilities & control mode (move: arrow & LShift, attack: Z & X) , eagerly waiting for merging. 

### PR DESCRIPTION
**Pull request by BAN-43-32532**
This is an experimental unit Udongein (happy New Year of the Rabbit) which adopts a common way to move (WASD or Arrow keys instead of as a tank), and it support two skills (I would rather call it Spell). It is an extension from a more basic unit SpellCaster. I would continue working on it, and I need others' suggestions.

### CODE REVIEW

File: src/battle_game/core/units/spell_caster.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  void BoundaryBetweenWavesAndParticles();
+  struct {
+    uint32_t clock_{0};
+  } directional_fire;
+  struct {
+    uint32_t clock_{0};
+    float angle_{0.0f};
+    float angle_increment_{glm::radians(1.0f)};
+  } boundary_between_waves_and_particles;
+};
```
Comment (LazyJazz) [inline src/battle_game/core/units/spell_caster.h:50]: Put these into new files

**Review by LazyJazz — CHANGES_REQUESTED** [review]
Good idea.

**Review by BAN-43-32532 — COMMENTED** [review]
I have made necessary changes, together with some utility updates.

**Comment by BAN-43-32532** [comment]
What should I do after completing these changes?

**Review by LazyJazz — DISMISSED** [review]
LGTM!

**Review by LazyJazz — APPROVED** [review]
LGTM!

**Pull request merged**